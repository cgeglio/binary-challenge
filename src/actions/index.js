export const login = user => ({
  type: 'LOGIN',
  user
})

export const logout = user => ({
  type: 'LOGOUT',
  user
})

export const addCards = cards => ({
  type: 'ADD_CARDS',
  cards
})

export const removeCards = cards => ({
  type: 'REMOVE_CARDS',
  cards
})

export const addQuestion = question => ({
  type: 'ADD_QUESTION',
  question
})

export const removeQuestion = question => ({
  type: 'REMOVE_QUESTION',
  question
})

export const addFortune = fortune => ({
  type: 'ADD_FORTUNE',
  fortune
})

export const removeFortune = fortune => ({
  type: 'REMOVE_FORTUNE',
  fortune
})

export const addReading = currentReading => ({
  type: 'ADD_READING',
  currentReading
})

export const removeReading = currentReading => ({
  type: 'REMOVE_READING',
  currentReading
})

export const addFavorite = favorite => ({
  type: 'ADD_FAVORITE',
  favorite
})

export const removeFavorite = favorite => ({
  type: 'REMOVE_FAVORITE',
  favorite
})

export const resetFavorites = favorites => ({
  type: 'RESET_FAVORITES',
  favorites
})
