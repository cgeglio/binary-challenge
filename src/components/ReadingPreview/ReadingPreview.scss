@import url('https://fonts.googleapis.com/css?family=Croissant+One&display=swap');
@import "../../mixins.scss";

.saved-preview {
  @include flex();
  position: relative;
}

.preview-cards {
  display: flex;
  flex-direction: row;
  height: 200px;
  overflow: hidden;
}

.preview-cards img {

}

.preview-question {
  color: black;
  margin-top: 90px;
  position: absolute;
}

.preview-overlay {
  align-self: center;
  background-color: #D5A1EE;
  height: 90%;
  opacity: .3;
  position: absolute;
  transition: .5s ease;
  width: 782px;
}

.view-reading-btn {
  @include buttons();
  bottom: 40px;
  height: 28px;
  left: 295px;
  position: relative;
  width: 160px;
    &:hover {
      background-color: white;
      color: black;
    }
}
