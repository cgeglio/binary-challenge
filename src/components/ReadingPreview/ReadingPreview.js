import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './ReadingPreview.scss';
import { addReading } from '../../actions';
import { connect } from 'react-redux'

export class ReadingPreview extends Component {

  updateCurrentReading = () => {
    this.props.makeSavedReadingCurrentReading(this.props.reading);
  }

  render() {
    return (
      <article className='saved-preview'>
        <div className='preview-cards'>
          {this.props.reading.cards.map(card => {
            return <><div className='preview-overlay'></div><img src={process.env.PUBLIC_URL + `/cardAssets/${card.name_short}.jpg`} alt="Tarot card" key={card.name} /></>
          })
        }
        </div>
        <h2 className='preview-question'>{this.props.reading.question}</h2>
        <Link to={`/saved/${this.props.reading.id}`}><button onClick={() => this.updateCurrentReading()} className='view-reading-btn'>View Reading</button></Link>
      </article>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  makeSavedReadingCurrentReading: savedReading => (dispatch(addReading(savedReading)))
})

export default connect(null, mapDispatchToProps)(ReadingPreview)
