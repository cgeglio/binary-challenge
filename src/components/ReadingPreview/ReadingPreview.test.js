import React from 'react';
import { ReadingPreview, mapDispatchToProps } from './ReadingPreview';
import { addReading } from '../../actions';
import { shallow } from 'enzyme';

describe('ReadingPreview', () => {

  let wrapper;
  let mockReading;
  let mockState;

  beforeEach(() => {
    mockReading = {cards: [{name: 'The Moon'}], question: 'Is bithcuits a bad kitty?', fortune: 'Cheaters never prosper', id: 30};
    wrapper = shallow(<ReadingPreview reading={mockReading} />);
  });

  describe('ReadingPreview container/component', () => {
    it ('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call makeSavedReadingCurrentReading when updateCurrentReading is invoked', () => {
      let mockSavedReading = jest.fn().mockImplementation();
      wrapper = shallow(<ReadingPreview reading={mockReading} makeSavedReadingCurrentReading={mockSavedReading} />);

      wrapper.instance().updateCurrentReading(mockReading);
      expect(mockSavedReading).toHaveBeenCalledWith(mockReading);
    });

    it('should call updateCurrentReading when view reading button is clicked', () => {
      wrapper.instance().updateCurrentReading = jest.fn();
      wrapper.instance().forceUpdate()
      wrapper.find('.view-reading-btn').simulate('click')
      expect(wrapper.instance().updateCurrentReading).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the addReading action when updateCurrentReading is called', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = addReading(mockReading);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.makeSavedReadingCurrentReading(mockReading);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });
});
