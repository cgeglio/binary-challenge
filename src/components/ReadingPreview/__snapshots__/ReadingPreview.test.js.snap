// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadingPreview ReadingPreview container/component should match the snapshot 1`] = `
<article
  className="saved-preview"
>
  <div
    className="preview-overlay"
  />
  <div
    className="preview-cards"
  >
    <div
      key="The Moon"
    >
      <img
        alt="Tarot card"
        key="The Moon"
        src="/cardAssets/undefined.jpg"
      />
    </div>
  </div>
  <h2
    className="preview-question"
  >
    Is bithcuits a bad kitty?
  </h2>
  <Link
    to="/saved/30"
  >
    <button
      className="view-reading-btn"
      onClick={[Function]}
    >
      View Reading
    </button>
  </Link>
</article>
`;
